File: Code Forces/PY/Contest.py
a,b,c,d = map(int, input().split())

misha = max((3*a)/10 , a-(a/250) * c)
vasya = max((3*b)/10 , b-(b/250) * d)

if vasya == misha: print("Tie")
elif vasya>misha: print("Vasya")
else: print("Misha")

File: Code Forces/CPP/Contest.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
	int a,b,c,d;
	cin>>a>>b>>c>>d;
	int misha = max((3*a)/10 , a-(a/250) * c);
	int vasya = max((3*b)/10 , b-(b/250) * d);

	if (vasya == misha)
		printf("Tie\n");
	else if(vasya>misha)
		printf("Vasya\n");
	else
		printf("Misha\n");


	return 0;
}